src/app/configuration/communities/community-list/community-list.component.ts

```diff
@@ -10,10 +18,61 @@ import { defaultColumnDefs } from './communities-table-data';
   templateUrl: './community-list.component.html',
   styleUrls: ['./community-list.component.scss'],
 })
-export class CommunityListComponent {
+export class CommunityListComponent implements OnDestroy {
   defaultColumns: TableColumn[] = defaultColumnDefs;
   actionsBulkEdit: NavItem[] = actionsBulkEditDefs;
   tableAppearanceOptions = TableAppearanceOptions;
 
-  constructor(public communityService: CommunityService) {}
+  private subscription: Subscription[] = [];
+
+  constructor(
+    public communityService: CommunityService,
+    private dialog: MatDialog,
+    private translateService: TranslateService,
+    private formActionsService: FormActionsService
+  ) {}
+
+  openEditCommunityModal(community: CommunityDto): void {
+    const dialogRef = this.dialog.open(AddEditBaseModalComponent, {
+      maxWidth: ADD_MODAL_MAX_WIDTH,
+      data: {
+        title: this.translateService.instant('headingEditCommunity'), // todotranslate
+        component: CommunityAddEditComponent,
+        resource: community,
+        archive: true,
+      },
+    });
+
+    this.subscription.push(
+      dialogRef.afterClosed().subscribe((result) => {
+        this.formActionsService.setDiscard();
+        if (result) {
+          // this.tableComponent.getResources(true);
+        }
+      })
+    );
+  }
+
+  openAddCommunityModal(): void {
+    const dialogRef = this.dialog.open(AddEditBaseModalComponent, {
+      maxWidth: ADD_MODAL_MAX_WIDTH,
+      data: {
+        title: this.translateService.instant('actionAddNewCommunity'), // todotranslate
```

Comment:
add the translation please

Reply:
It is added, just need to delete the todotranslate what I forgot.